### DISCUSSION TITLE
Handling Mesh Mapping for Faces Between Different Meshes in MOOSE

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#26446] [Q&A Modules: General] [answered]
I've learned about the existence of CoarseMeshExtraElementIDGenerator in MOOSE, which seems to be designed for mapping between meshes in different layers. However, it appears that it only stores the mapping between cells using ExtraElementID. I'm wondering if MOOSE provides a method to handle mapping between faces of different meshes. Is there a functionality or another class that can be used for this purpose? Any guidance or information on this matter would be greatly appreciated.

**Comment by GiudGiud** [comment, marked as answer]
Hello

We dont have anything out of the box. However you could:

- redevelop the CoarseMeshExtra....IDGenerator with Nodal extra IDs instead of element IDs. Then the nodes are on faces and you can work with that
OR
- from the fine face, you can get the fine element near the face, then the coarse element near the face using the ID, then from there you could find the coarse element face itself. It's a little convoluted but this is not going to be very expensive to do.

Once you have created the mapping, you can cache it if you care about performance

Guillaume

**Reply by hityyds** [reply]
I get it, thank you for your response.
